typescript/sdk/src/core/EvmCoreModule.ts

```diff
@@ -90,20 +93,136 @@ export class EvmCoreModule extends HyperlaneModule<
+   * @param expectedConfig - The expected token router configuration, including the ISM configuration.
+   * @returns Transaction that need to be executed to update the ISM configuration.
+   */
+  async createDefaultIsmUpdateTxs(
+    actualConfig: CoreConfig,
+    expectedConfig: CoreConfig,
+  ): Promise<AnnotatedEV5Transaction[]> {
+    const updateTransactions: AnnotatedEV5Transaction[] = [];
+
+    const actualDefaultIsmConfig = actualConfig.defaultIsm as DerivedIsmConfig;
+
+    // Try to update (may also deploy) Ism with the expected config
+    const { deployedIsm, ismUpdateTxs } = await this.deployOrUpdateIsm(
+      actualDefaultIsmConfig,
+      expectedConfig.defaultIsm,
+    );
+
+    if (ismUpdateTxs.length) {
+      updateTransactions.push(...ismUpdateTxs);
+    }
+
+    const newIsmDeployed = actualDefaultIsmConfig.address !== deployedIsm;
+    if (newIsmDeployed) {
+      const { mailbox } = this.serialize();
+      const contractToUpdate = Mailbox__factory.connect(
+        mailbox!,
+        this.multiProvider.getProvider(this.domainId),
+      );
+      updateTransactions.push({
+        annotation: `Setting default ISM for Mailbox ${mailbox!} to ${deployedIsm}`,
+        chainId: this.domainId,
+        to: contractToUpdate.address,
+        data: contractToUpdate.interface.encodeFunctionData('setDefaultIsm', [
+          deployedIsm,
+        ]),
+      });
+    }
+
+    return updateTransactions;
+  }
+
+  /**
+   * Updates or deploys the ISM using the provided configuration.
+   *
+   * @returns Object with deployedIsm address, and update Transactions
+   */
+  public async deployOrUpdateIsm(
+    actualDefaultIsmConfig: DerivedIsmConfig,
+    expectDefaultIsmConfig: IsmConfig,
+  ): Promise<{
+    deployedIsm: Address;
+    ismUpdateTxs: AnnotatedEV5Transaction[];
+  }> {
+    const {
+      mailbox,
+      staticAggregationIsmFactory,
+      staticMerkleRootMultisigIsmFactory,
+      staticMessageIdMultisigIsmFactory,
+      staticAggregationHookFactory,
+      domainRoutingIsmFactory,
+    } = this.serialize();
+
+    assert(mailbox, 'mailbox is undefined in this.args.addresses');
+    assert(
+      staticAggregationIsmFactory,
+      'staticAggregationIsmFactory is undefined in this.args.addresses',
+    );
+    assert(
+      staticMerkleRootMultisigIsmFactory,
+      'staticMerkleRootMultisigIsmFactory is undefined in this.args.addresses',
+    );
+    assert(
+      staticMessageIdMultisigIsmFactory,
+      'staticMessageIdMultisigIsmFactory is undefined in this.args.addresses',
+    );
+    assert(
+      staticMessageIdMultisigIsmFactory,
+      'staticMessageIdMultisigIsmFactory is undefined in this.args.addresses',
+    );
+    assert(
+      domainRoutingIsmFactory,
+      'domainRoutingIsmFactory is undefined in this.args.addresses',
+    );
+
+    assert(
+      staticAggregationHookFactory,
+      'staticAggregationHookFactory is undefined in this.args.addresses',
+    );
```

Comment:
why is this function checking the output of this.serialize?

Reply:
Removed